pipelines/release/Jenkinsfile.stop

```diff
@@ -0,0 +1,177 @@
+                            returnStdout: true
+                            ).trim().split('\\.git')[0]+'/pulls')
+                        }
+                    }
+                    // Unknown error
+                    else {
+                        error "Could not raise the PR\n"+sh(
+                        script: 'cat /tmp/output',
+                        returnStdout: true
+                        ).trim()
+                    }
+                }
+            }
+        }
+        stage("Merging master with ${params.releaseBranch}"){
+
+            if( sh (
+                script: " git checkout origin/master && git merge origin/${params.releaseBranch} ",
+                returnStatus: true
+                ) != 0){
+                    ansiColor('xterm'){
+                        println(ANSI_BOLD + ANSI_RED +
+                        "Merge Conflict\nPR Raised\nPlease fix the Conflicts and run the job again\nPR: "
+                        +ANSI_NORMAL+sh(
+                        script: 'git config --get remote.origin.url',
+                        returnStdout: true
+                        ).trim().split('\\.git')[0]+'/pulls')
+                    }
+                    error 'Merge Conflict'
+            }
+        }
+        stage('pushing tag to upstream'){
+            // Using withCredentials as gitpublish plugin is not yet ported for pipelines
+            // Defining credentialsId for default value passed from Parameter or environment value.
+            // gitCredentialId is username and password type
+            withCredentials([usernamePassword(credentialsId: "${gitCredentialId}",
+            passwordVariable: 'gitPassword', usernameVariable: 'gitUser')]) {
+
+                // Getting git remote url
+                origin = "https://${gitUser}:${gitPassword}@"+sh (
+                script: 'git config --get remote.origin.url',
+                returnStdout: true
+                ).trim().split('https://')[1]
+                echo "Git Hash: ${origin}"
+                // Checks whether remtoe branch is present
+                // Stdouts 1 if true
+                remoteBranch =
+                ansiColor('xterm'){
+                    // If remote tag exists
+                    if( sh(script: "git ls-remote --tags --exit-code ${origin} ${params.releaseBranch}", returnStatus: true) == 0 ) {
+                        println(ANSI_BOLD + ANSI_RED + "Upstream has tag with same name: ${params.releaseBranch}\nPR: "
+                        +ANSI_NORMAL+sh(
+                            script: 'git config --get remote.origin.url',
+                            returnStdout: true
+                        ).trim().split('\\.git')[0]+'/pulls')
+                        error 'remote tag found with same name'
+                    }
+                }
+                // Pushing tag
+                sh("git push ${origin} HEAD:tags/${params.releaseBranch}")
```

Comment:
A better order of code would be - First create tag and push, then merge. If conflicts, raise PR and exit. If no conflicts, auto merge, delete branch and exit.